File: sample.ts
// sample.ts
class Person {
    name: string;
    constructor(name: string) {
        this.name = name;
    }
    introduce(): string {
        return `Hi, I'm ${this.name}`;
    }
}

class MathService {
    multiply(a: number, b: number): number {
        return a * b;
    }
    divide(a: number, b: number): number {
        if (b === 0) throw new Error("Divide by zero");
        return a / b;
    }
}

function sayHello(name: string): string {
    return `Hello, ${name}`;
}

function factorialTS(n: number): number {
    if (n <= 1) return 1;
    return n * factorialTS(n - 1);
}

File: sample.py
# sample.py
class Calculator:
    def __init__(self, value=0):
        self.value = value
    def add(self, num):
        self.value += num
        return self.value
    def multiply(self, num):
        self.value *= num
        return self.value

class User:
    def __init__(self, name):
        self.name = name
    def login(self, password):
        if password == "1234":
            return True
        return False

def greet(name):
    return f"Hello, {name}!"
def factorial(n):
    if n == 0:
        return 1
    return n * factorial(n - 1)
